projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-checkout.ts

```diff
@@ -74,10 +61,15 @@ export function addB2bProductToCartAndCheckout() {
 }
 
 export function enterPONumber() {
+  const getPaymentTypes = interceptPaymentTypesEndpoint();
+
+  cy.wait(`@${getPaymentTypes}`).its('response.statusCode').should('eq', 200);
```

Comment:
This will always be flaky as there's going to have a race condition at times. Recent commits 'checks' shows the flakiness of the e2e now

This `wait` should be when we click the button to start the checkout process, for example, clicking the 'proceed to checkout button'. 

Should be within this [function](https://github.com/SAP/spartacus/blob/develop/projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/checkout/b2b-account-checkout-flow.core-e2e-spec.ts#L29)

Reply:
P.S

for flaky test, you should run them in headless like 10-20 times to verify if it actual got fixed, and like maybe ~5 times in CI.

Locally, you can include this in your root `package.json`. Make sure your cypress.json is pointing to b2b and that your actual application is on b2b.
> projects/storefrontapp-e2e-cypress/cypress/integration/b2b/regression/checkout/b2b-account-checkout-flow.core-e2e-spec.ts
```
testcy": "for i in {1..20}; do yarn e2e:run --spec cypress/integration/b2b/regression/checkout/b2b-account-checkout-flow.core-e2e-spec.ts; done",
```

Reply:
This is very good point! I was wondering about this problem over the weekend before I saw your comment and came to the same conclusions - we have to run this `wait` beforehand, because there is a chance that by checking it in the current step the data is already downloaded, so we intercepting the query that has already been made before.